air/src/stack/op_flags/mod.rs

```diff
@@ -76,405 +356,530 @@ impl<E: FieldElement> OpFlags<E> {
     #[inline(always)]
     pub fn split(&self) -> E {
         self.degree6_op_flags[get_op_index(Operation::Split.op_code())]
     }
 
+    /// Operation Flag of LOOP operation.
     #[inline(always)]
     pub fn loop_op(&self) -> E {
         self.degree6_op_flags[get_op_index(Operation::Loop.op_code())]
     }
 
     // ------ Degree 4 stack operations  -----------------------------------------------------
 
+    /// Operation Flag of MRUPDATE operation.
     #[inline(always)]
     pub fn mrupdate(&self) -> E {
         self.degree4_op_flags[get_op_index(Operation::MrUpdate(true).op_code())]
     }
 
+    /// Operation Flag of PUSH operation.
     #[inline(always)]
     pub fn push(&self) -> E {
         self.degree4_op_flags[get_op_index(Operation::Push(ONE).op_code())]
     }
 
+    /// Operation Flag of END operation.
     #[inline(always)]
     pub fn end(&self) -> E {
         self.degree4_op_flags[get_op_index(Operation::End.op_code())]
     }
 
+    /// Operation Flag of REPEAT operation.
     #[inline(always)]
     pub fn repeat(&self) -> E {
         self.degree4_op_flags[get_op_index(Operation::Repeat.op_code())]
     }
 
+    /// Operation Flag of RESPAN operation.
     #[inline(always)]
     pub fn respan(&self) -> E {
         self.degree4_op_flags[get_op_index(Operation::Respan.op_code())]
     }
 
+    /// Operation Flag of HALT operation.
     #[inline(always)]
     pub fn halt(&self) -> E {
         self.degree4_op_flags[get_op_index(Operation::Halt.op_code())]
     }
+
+    // ------ Composite Flags ---------------------------------------------------------------------------
+
+    // ------ No shift composite flags ------------------------------------------------------------------
+
+    /// Returns the composite flag when the element at depth index in the stack remains the same during
+    /// an operation.
+    #[inline(always)]
+    pub fn no_shift(&self, index: usize) -> E {
+        self.no_shift_flags[index]
+    }
+
+    /// Returns the composite flag when the element at depth index in the stack shifts to the left during
+    /// an operation. The left shift is not defined on the first postion in the stack and therefore, a ZERO
+    /// will be returned.
+    #[inline(always)]
+    pub fn left_shift(&self, index: usize) -> E {
+        self.left_shift_flags[index]
+    }
+
+    /// Returns the composite flag when the element at depth index in the stack shifts to the right during
+    /// an operation. The left shift is not defined on the first postion in the stack and therefore, a ZERO
+    /// will be returned.
+    #[inline(always)]
+    pub fn right_shift(&self, index: usize) -> E {
+        self.right_shift_flags[index]
+    }
+
+    // --------------------------- other composite flags -----------------------------------------
+
+    /// Returns the flag when the stack operation shifts the flag to the right.
+    #[inline(always)]
+    pub fn right_shift_flag(&self) -> E {
```

Comment:
nit: let's rename this to just `right_shift` and same for other similar methods.